server/src/main/java/org/elasticsearch/cluster/block/ClusterBlocks.java

```diff
@@ -68,27 +67,37 @@ public Map<String, Set<ClusterBlock>> indices(ClusterBlockLevel level) {
     }
 
     private Set<ClusterBlock> blocksForIndex(ClusterBlockLevel level, String index) {
-        return indices(level).getOrDefault(index, emptySet());
+        return indices(level).getOrDefault(index, Set.of());
     }
 
     private static EnumMap<ClusterBlockLevel, ImmutableLevelHolder> generateLevelHolders(
         Set<ClusterBlock> global,
         Map<String, Set<ClusterBlock>> indicesBlocks
     ) {
         EnumMap<ClusterBlockLevel, ImmutableLevelHolder> levelHolders = new EnumMap<>(ClusterBlockLevel.class);
+        List<ClusterBlock> setBuilder = new ArrayList<>();
+        Map<String, Set<ClusterBlock>> indicesBuilder = Maps.newMapWithExpectedSize(indicesBlocks.size());
         for (final ClusterBlockLevel level : ClusterBlockLevel.values()) {
```

Comment:
This seems to iterate both `global` and `indicesBlocks` 4 times (for each `ClusterBlockLevel`).
Would it be beneficial to iterate once collecting to enum maps per level and then merge them into a final representation?

Reply:
Not really right? It's still the same amount of work isn't it? I guess it's technically cheaper to iterate the enum instead of the map though ... but I think were talking about a relatively small gain here.